tests/suites/coslite/cl.sh

```diff
@@ -9,36 +9,68 @@ run_deploy_coslite() {
-
-	echo "Wait for all unit agents to be in active condition"
-	wait_for 0 "$(not_idle_list)" 1800
-
-	echo "Check that all offer endpoints specified in the overlays exist"
-	wait_for 5 '[.offers[] | .endpoints] | length'
+	juju deploy cos-lite --trust --channel=stable
+	echo "Wait for all unit agents to be in idle condition"
+	wait_for 0 "$(idle_list)" 1800
 
+	# run-action will change in 3.0
 	admin_passwd=$(juju run-action --wait grafana/0 get-admin-password --format json | jq '.["unit-grafana-0"]["results"]["admin-password"]')
 	if [ -z "$admin_passwd" ]; then
 		echo "expected to get admin password for grafana/0"
 		exit 1
 	fi
 
+	assert_dashboards
+	echo "cos lite tests passed"
+	# TODO(basebandit): enable destroy-model once model teardown has been fixed for k8s models.
+	#destroy_model "${model_name}"
+}
+
+assert_dashboards() {
 	# Assert the web dashboards are reachable
-	alertmanager_ip=$(juju status --format json | jq '.applications["alertmanager"]["units"]["alertmanager/0"].address' | tr -d '"')
+	traefik_ip=$(juju status --format json | jq '.applications["traefik"]["units"]["traefik/0"].address' | tr -d '"')
+	get_proxied_unit_url "prometheus" 0
+	prometheus_url_path=$(get_proxied_unit_url "prometheus" 0 | awk -F / '{print "/"$NF}')
+	echo "check if prometheus is ready to serve traffic"
+	# Replace the external, metallb URL with the one of traefik, to skip a lot of routing 'fun' (avoids tests failing intermittently)
+	check_ready http://"$traefik_ip":80"$prometheus_url_path"/-/ready 200
+
 	echo "Check if alertmanager is ready to serve traffic"
-	curl -sS http://"$alertmanager_ip":9093/ready -o /dev/null
-	grafana_ip=$(juju status --format json | jq '.applications["grafana"]["units"]["grafana/0"].address' | tr -d '"')
+	alertmanager_ip=$(juju status --format json | jq '.applications["alertmanager"]["units"]["alertmanager/0"].address' | tr -d '"')
+	# TODO(basebandit): Change this when AM is exposed over the ingress
+	check_ready http://"$alertmanager_ip":9093 200
+
 	echo "Check if grafana is ready to serve traffic"
-	curl -sS http://"$grafana_ip":3000/api/health -o /dev/null
-	prometheus_ip=$(juju status --format json | jq '.applications["prometheus"]["units"]["prometheus/0"].address' | tr -d '"')
-	echo "check if prometheus is ready to serve traffic"
-	curl -sS http://"$prometheus_ip":9090/ready -o /dev/null
-	echo "cos lite tests passed"
+	grafana_ip=$(juju status --format json | jq '.applications["grafana"]["units"]["grafana/0"].address' | tr -d '"')
+	check_ready http://"$grafana_ip":3000 200
+}
 
-	# without --force grafana get stuck in a hook(removal) error state.
-	juju remove-application grafana --force
+get_proxied_unit_url() {
+	local unit unit_num
+	unit=${1}
+	unit_num=${2}
+	proxied_endpoints=$(juju run-action --wait traefik/0 show-proxied-endpoints --format json | jq -r '.[] | .results["proxied-endpoints"]')
+	echo "$proxied_endpoints" | jq ".[\"$unit/$unit_num\"][\"url\"]" | tr -d '"'
```

Comment:
Also, is there a reason this is broken up into multiple lines in this way? I'm not necessarily against it for readability, but I suspect it may be better on a single line